x-pack/plugins/apm/public/components/app/service_map/Popover/backend_contents.tsx

```diff
@@ -57,12 +57,14 @@ export function BackendContents({
   );
 
   const isLoading = status === FETCH_STATUS.LOADING;
-  const detailsUrl = apmRouter.link('/backends/overview', {
-    query: {
-      ...query,
-      backendName,
-    } as TypeOf<ApmRoutes, '/backends/overview'>['query'],
-  });
+  const detailsUrl = backendName
+    ? apmRouter.link('/backends/overview', {
+        query: {
+          ...query,
+          backendName,
+        } as TypeOf<ApmRoutes, '/backends/overview'>['query'],
+      })
+    : undefined;
```

Comment:
Why didn't this fail before? Is `backendName` optional when it should be required?

Reply:
`backend` is required for the `/backends/overview` route, but not all nodes on service maps have it, so the problem was that it was trying to generate the URL with an undefined value.